### PR TITLE
Aus CMP 0% test

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

puts AUS cmp behind 0% test

### todo
- [x] install new CMP version once https://github.com/guardian/consent-management-platform/pull/303 merges

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes https://github.com/guardian/dotcom-rendering/pull/1991

### CODE REVIEW

File: common/app/experiments/Experiments.scala [hunk 1]
```diff
@@ -13,6 +13,7 @@ object ActiveExperiments extends ExperimentsDefinition {
     DCRBubble,
     NGInteractiveDCR,
     NewsletterEmbedDesign,
+    UseAusCmp
```
Comment (mxdvl) [inline common/app/experiments/Experiments.scala:16]: Can we have a trailing `,` comma?

File: common/app/experiments/Experiments.scala [hunk 2]
```diff
@@ -64,3 +65,13 @@ object NewsletterEmbedDesign
       sellByDate = new LocalDate(2020, 11, 2),
       participationGroup = Perc0C,
     )
+
+object UseAusCmp
+    extends Experiment(
+      name = "use-aus-cmp",
+      description = "Use AU framework in CMP in Australia",
+      owners = group(Commercial),
+      sellByDate = new LocalDate(2020, 11, 6),
```
Comment (mxdvl) [inline common/app/experiments/Experiments.scala:74]: How is this decided?

File: static/src/javascripts/boot.js [hunk 3]
```diff
@@ -38,7 +39,18 @@ const go = () => {
...
         const pubData: { browserId?: ?string, pageViewId?: ?string } =
             { browserId, pageViewId };
-        cmp.init({ pubData, isInUsa: isInUsa() });
+
+
+        if (isInAustralia()) {
+            if (config.get('tests.useAusCmpVariant') === 'variant') {
+                cmp.init({ pubData, country: 'AU' });
+            } else {
+                cmp.init({ pubData, country: 'GB' });
```
Comment (mxdvl) [inline static/src/javascripts/boot.js:48]: Make it explicit, why we are setting GB in AU.

```suggestion
                cmp.init({ pubData, country: 'GB' }); // Use TCFv2 Framework
```

**Review by mxdvl — COMMENTED** [review]
This looks good, but of course is lacking the new package.

File: static/src/javascripts/boot.js [hunk 4]
```diff
@@ -10,8 +10,9 @@ import { markTime } from 'lib/user-timing';
 import { captureOphanInfo } from 'lib/capture-ophan-info';
 import reportError from 'lib/report-error';
 import { cmp } from '@guardian/consent-management-platform';
-import { isInUsa } from 'projects/common/modules/commercial/geo-utils.js';
 import { getCookie } from 'lib/cookies';
+import { isInAustralia } from 'common/modules/commercial/geo-utils';
```
Comment (ioanna0) [inline static/src/javascripts/boot.js:14]: we sometimes consider New Zealand as well when doing regional stuff in AUS, wondering if its the same for CMP?

File: common/app/experiments/Experiments.scala (hunk 1 above)
Comment (sndrs) [reply common/app/experiments/Experiments.scala:16]: not in scala

File: common/app/experiments/Experiments.scala (hunk 2 above)
Comment (sndrs) [reply common/app/experiments/Experiments.scala:74]: i chose it, but it seems a good time away?

File: common/app/experiments/Experiments.scala (hunk 1 above)
Comment (mxdvl) [reply common/app/experiments/Experiments.scala:16]: How comes the line above had one?

File: static/src/javascripts/boot.js (hunk 4 above)
Comment (mxdvl) [reply static/src/javascripts/boot.js:14]: Worth checking with stakeholders!

Comment (sndrs) [reply static/src/javascripts/boot.js:14]: i _think_ this is Australia-only

File: common/app/experiments/Experiments.scala (hunk 1 above)
Comment (sndrs) [reply common/app/experiments/Experiments.scala:16]: oooh that is an EXCELLENT point!

Comment (sndrs) [reply common/app/experiments/Experiments.scala:16]: an you know what, as i replied i thought "i sound far more confident here than i should/am" 🤦

Comment (mxdvl) [reply common/app/experiments/Experiments.scala]: And this seems to have been the reason for the build failure, too?

File: package.json [hunk 5]
```diff
@@ -15,7 +15,7 @@
     "@guardian/automat-modules": "^0.3.7",
     "@guardian/braze-components": "0.0.12",
     "@guardian/commercial-core": "^0.9.0",
-    "@guardian/consent-management-platform": "6.4.0",
+    "@guardian/consent-management-platform": "7.0.3",
```
Comment (mxdvl) [inline package.json:18]: [The future is now](https://github.com/guardian/consent-management-platform/releases/tag/v7.0.4)!

```suggestion
    "@guardian/consent-management-platform": "7.0.4",
```

**Review by mxdvl — APPROVED** [review]

File: static/src/javascripts/boot.js [hunk 6]
```diff
@@ -38,7 +39,12 @@ const go = () => {
         const pageViewId: ?string = config.get('ophan.pageViewId');
         const pubData: { browserId?: ?string, pageViewId?: ?string } =
             { browserId, pageViewId };
-        cmp.init({ pubData, isInUsa: isInUsa() });
+
+        if (config.get('tests.useAusCmpVariant') === 'variant') {
```
Comment (sndrs) [inline static/src/javascripts/boot.js:43]: @mxdvl simplified this logic per our chat

**Review by mxdvl — APPROVED** [review]
Yes, it makes sense, now! The test is for the “new API”, which is required to enable Australia.

File: static/src/javascripts/boot.js (hunk 6 above)
Comment (mxdvl) [reply static/src/javascripts/boot.js:43]: For people outside the test, the old interface is used, still.

**Pull request merged**